docs/enhancements/fep-001-api-fleet-templating.md

```diff
@@ -0,0 +1,303 @@
+# EP: Definition of the Fleet and Device API spec
+<!-- this format is inspired by the K8S KEP format https://raw.githubusercontent.com/kubernetes/enhancements/master/keps/NNNN-kep-template/README.md -->
+## Summary
+
+The Fleet API is designed to manage big groups of devices that share configuration
+and operating system image.
+
+Some use cases point to the necessity of customizing some of those details based
+on the device's metadata.
+
+## Motivation
+
+Allowing templating of the device spec in the fleet API we could achieve a more
+flexible management, allowing customers to manage devices in a more granular way.
+
+### Goals
+
+* Define templating mechanism for the device spec template configuration in fleets,
+  based on labels and other metadata.
+
+* Enable different os images based on the device labels.
+```yaml
+  template:
+    spec:
+      os:
+        image: "access-control-{{ index .device.metadata.labels `region` }}:1.0.0"
+```
+
+* Enable different references for a config source based on labels:
+```yaml
+- name: microshift-manifests
+  configType: GitConfigProviderSpec
+  gitRef:
+    repository: basic-nginx-demo
+    targetRevision: "1.0.0-{{ index .device.metadata.labels `region` }}"
```

Comment:
Can we limit targetRevision to have a single parameter?  We need to map the parameter to frozen hash, so this will be easier without multiple parameters.